lib/dry/types/default.rb

```diff
@@ -23,7 +24,7 @@ def evaluate
       alias_method :evaluate, :value
 
       # @param [Object, #call] value
-      # @return [Default, Dry::Types::Default::Callable]
+      # @return [Class] `Default` or `Callable`
```

Comment:
> Unfortunately, AFAIK YARD does not support putting specific Class instances returned from a method in @return tag (because the class name in @return block means “instance of the class”).

Yeah. I'd like some form of syntax for this eventually. I opened https://github.com/backus/yardcheck/issues/6 so I can start thinking about solutions for Yardcheck